commandhandler.go

```diff
@@ -0,0 +1,556 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/acarl005/stripansi"
+	"github.com/shurcooL/tictactoe"
+)
+
+type commandInfo struct {
+	name          string
+	description   string
+	callable      func(user *user, args []string)
+	echo          bool
+	requiresAdmin bool
+	aliases       []string
+}
+
+var (
+	commands = make([]commandInfo, 0)
+)
+
+func processMessage(u *user, message string) {
+	message = strings.TrimSpace(message)
+
+	if message == "" {
+		return // Don't allow empty messages
+	}
+	splitted := strings.Split(message, " ")
+	if strings.HasPrefix(splitted[0], "./") {
+		if u.slack {
+			u.room.broadcast(devbot, "Slack users can't use commands", false)
+			return
+		}
+		commandName := strings.TrimPrefix(splitted[0], "./")
+		for _, command := range commands {
+			if command.name == commandName {
+				// Command found
+				runCommand(u, command, splitted, message)
```

Comment:
don't pass in a split string because most commands don't split anyways

Reply:
Most do actually.

Reply:
./nick, ./room, ./tic, ./tz, ./hang, ./id, ./ban, ./kick don't

Reply:
and those are all the commands with args

Reply:
./ban ./tz ./room and ./nick does

Reply:
no?
For example this is how ban gets the arg:
```go
...
findUserByName(u.room, strings.TrimSpace(strings.TrimPrefix(line, "./ban")))
...
```

Reply:
rest do the same

Reply:
So ya please change this (not the highest priority)

Reply:
@TAG-Epic don’t resolve until we are at a consensus

Reply:
@TAG-Epic we need this done